apps/rush-lib/src/cli/actions/ChangeAction.ts

```diff
@@ -311,26 +316,39 @@ export class ChangeAction extends BaseRushAction {
   }
 
   private _getChangedPackageNames(): string[] {
-    const changedFolders: (string | undefined)[] | undefined = VersionControl.getChangedFolders(
+    const changedFiles: string[] = VersionControl.getChangedFiles(
       this._targetBranch,
       this._noFetchParameter.value
     );
-    if (!changedFolders) {
+    if (changedFiles.length === 0) {
       return [];
     }
     const changedPackageNames: Set<string> = new Set<string>();
 
     const repoRootFolder: string | undefined = VersionControl.getRepositoryRootPath();
     const projectHostMap: Map<string, string> = this._generateHostMap();
 
+    const getChangeIgnorePatterns: (fsPath: string) => string[] = memoize((fsPath: string): string[] => {
+      if ((repoRootFolder && Path.isUnder(repoRootFolder, fsPath)) || !FileSystem.exists(fsPath)) {
+        return [];
+      }
+      const parentIgnoreStrings: string[] = getChangeIgnorePatterns(path.dirname(fsPath));
+
+      const changeIgnoreFilePath: string = path.join(fsPath, RushConstants.changeignoreFileName);
+
+      return FileSystem.exists(changeIgnoreFilePath)
+        ? [...parentIgnoreStrings, FileSystem.readFile(changeIgnoreFilePath)]
+        : parentIgnoreStrings;
+    });
+
     this.rushConfiguration.projects
```

Comment:
I recommend inverting the lookup:
```ts
const changedFilesByProject: Map<RushConfigurationProject, string[]> = new Map();

for (const file of changedFiles) {
  const relevantProject: RushConfigurationProject | undefined = this.rushConfiguration.findProjectByPosixRelativePath(file);
  if (!relevantProject) {
    continue;
  }
  let projectFiles: string[] | undefined = changedFilesByProject.get(relevantProject);
  if (!projectFiles) {
    if (!project.shouldPublish || project.versionPolicy && project.versionPolicy.exemptFromRushChange) {
      continue;
    }
    projectFiles = [];
    changedFilesByProject.set(relevantProject, projectFiles);
  }

  // Potentially test the file here
  projectFiles.push(file);
}

for (const [project, relevantFiles] of changedFilesByProject) {
  // Do your ignorer config processing and filtering here, and find out what is left.
  // Might consider recording which files prompted the change for posterity, so that we can have a --verbose option or similar for rush change that logs which files required the change file.
}
```